Sasquatch/SasquatchObjC/AppDelegate.m

```diff
@@ -383,7 +389,33 @@ - (BOOL)distribute:(MSDistribute *)distribute
   return NO;
 }
 
-#pragma mark - MSPushDelegate
+#pragma mark - Push callbacks
+
+// iOS 10 and later, called when a notification is delivered to an app that is
+// in the foreground.
+// When this callback is called, this disables the other callback that MSPush
+// handles.
+- (void)userNotificationCenter:(UNUserNotificationCenter *)center
+       willPresentNotification:(UNNotification *)notification
+         withCompletionHandler:
+             (void (^)(UNNotificationPresentationOptions options))
+                 completionHandler API_AVAILABLE(ios(10.0)) {
+  self.didReceiveNotificationInForeground = YES;
+  [MSPush didReceiveRemoteNotification:notification.request.content.userInfo];
+  completionHandler(UNNotificationPresentationOptionNone);
```

Comment:
Let's expose it to UI to test system alert in the foreground case.

Reply:
What do you want to expose ?

Reply:
There is "Alert" option. In this case the callback is called twice - here and when user clicked

Reply:
So, either picker or just switch "show system notification on foreground push"

Reply:
I don't understand this feedback, can you rephrase?